src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,45 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `checkHealth` method is not the most efficient way to remove dead animals from the `Animal.alive` array. Instead of calling this method for each animal, it would be more efficient to filter the `Animal.alive` array directly inside the `bite` method after reducing the prey's health. This way, you avoid the overhead of multiple function calls and make the intention of the code clearer.